src/rdf4cpp/rdf/storage/tuple/IDatasetBackend.hpp

```diff
@@ -17,14 +17,14 @@ class IDatasetBackend {
     using QuadPattern = rdf4cpp::rdf::query::QuadPattern;
 
 protected:
-    mutable node::NodeStorage node_storage_;
+    mutable node::WeakNodeStorage node_storage_;
```

Comment:
The original idea behind Datasets having a NodeStorage was that the dataset keeps a NodeStorage alive.

Reply:
I guess that makes sense. The reason why i changed it to weak was that there is a default instance for the dataset storage which will keep effectively the default backend alive forever. If that is expected then I will change it back.

Reply:
I would however suggest to change the initialization behaviour of the DatasetStorage default instance to only get initialized when first accessed and not on program start

Reply:
is reverted